### PR TITLE
feat: truncate big urls in publication

### LINKED ISSUE #3072: Truncate big URLs in publication

### PR DESCRIPTION
**Pull request by foolo**
## What does this PR do?

Truncate length, and strip http(s) and www
Inspired from Twitter's truncation feature, which looks like this:
Screenshot_2023-06-13_19-08-11

### Before:

b Screenshot_2023-06-15_09-15-34
b Screenshot_2023-06-15_23-23-20

### After:
a Screenshot_2023-06-15_09-15-59
a Screenshot_2023-06-15_23-21-50

### <samp>🤖 Generated by Copilot at 5eec6ee</samp>

Added a `truncateUrl` function to the `@lenster/lib` package and used it to improve the display of external links in the `MarkupLink` component. Added tests for the function in `tests/packages/lib/truncateUrl.spec.ts`.

## Related issues

Fixes #3072 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at 5eec6ee</samp>

* Import and use `truncateUrl` function to shorten link titles in `MarkupLink` component ([link](https://github.com/lensterxyz/lenster/pull/3102/files?diff=unified&w=0#diff-86598e03b5162961a77d6337626e4e4d7501d6176353a87fdd69d873841ea798R1), [link](https://github.com/lensterxyz/lenster/pull/3102/files?diff=unified&w=0#diff-86598e03b5162961a77d6337626e4e4d7501d6176353a87fdd69d873841ea798L22-R23))
* Define and export `truncateUrl` function in `@lenster/lib` package to truncate and strip URLs ([link](https://github.com/lensterxyz/lenster/pull/3102/files?diff=unified&w=0#diff-ff70f7cadcf45f1ce23414a996feb86a7990b46a77fc881533df817961dd3cd4R1-R16))

## Emoji

📦🧪💄

### CODE REVIEW

**Comment by bigint** [comment]
@foolo we don't want to trucate lenster URLs IMO 🙇🏼 it looks bit bad

**Comment by foolo** [comment]
> @foolo we don't want to trucate lenster URLs IMO 🙇🏼 it looks bit bad

@bigint Ahh, good point. You mean anything on `*.lenster.xyz`, right?

**Comment by bigint** [comment]
@foolo exactly

**Comment by foolo** [comment]
> @foolo exactly

@bigint Done! I updated the screenshots in the description too.

File: apps/web/src/components/Shared/Markup/MarkupLink/index.tsx [hunk 1]
```diff
@@ -19,7 +20,7 @@ const MarkupLink = ({ href, title = href }: MarkupLinkProps) => {
     return <Hashtag href={href} title={title} />;
   }
 
-  return <ExternalLink href={href} title={title} />;
+  return <ExternalLink href={href} title={title && truncateUrl(title, 30)} />;
```
Comment (bigint) [inline apps/web/src/components/Shared/Markup/MarkupLink/index.tsx:23]: being title is not a boolean, i suggest to use ternary operator 🙇🏼

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Shared/Markup/MarkupLink/index.tsx (hunk 1 above)
Comment (foolo) [reply apps/web/src/components/Shared/Markup/MarkupLink/index.tsx]: Done!

File: packages/lib/truncateUrl.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Strip a url, truncate it to a given max length and add ellipsis if truncated.
+ * @param url url to truncate
```
Comment (bigint) [inline packages/lib/truncateUrl.ts:3]:
```suggestion
 * Strip a url, truncate it to a given max length and add ellipsis if truncated.
 *
 * @param url url to truncate
```

File: tests/packages/lib/truncateUrl.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import truncateUrl from '@lenster/lib/truncateUrl';
+import { expect, test } from '@playwright/test';
+
+test.describe('truncateUrl', () => {
+  test('should strip http(s) and www', () => {
+    const url = 'https://www.example.com/foo';
+    expect(truncateUrl(url, 30)).toEqual('example.com/foo');
+  });
+
+  test('truncate url that is longer than max length', () => {
```
Comment (bigint) [inline tests/packages/lib/truncateUrl.spec.ts:10]:
```suggestion
  test('should truncate url that is longer than max length', () => {
```

File: tests/packages/lib/truncateUrl.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const url = 'https://example.com/path?key=value';
+    expect(truncateUrl(url, 20)).toEqual('example.com/path?ke…');
+  });
+
+  test('do not truncate url that is max length or shorter (after prefix stripped)', () => {
```
Comment (bigint) [inline tests/packages/lib/truncateUrl.spec.ts:15]:
```suggestion
  test('should not truncate url that is max length or shorter (after prefix stripped)', () => {
```

File: tests/packages/lib/truncateUrl.spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const maxLengthUrl = 'https://example.com/pathname';
+    const shortUrl = 'https://example.com/foo';
+    expect(truncateUrl(maxLengthUrl, 20)).toEqual('example.com/pathname');
+    expect(truncateUrl(shortUrl, 20)).toEqual('example.com/foo');
+  });
+
+  test('do not truncate *.lenster.xyz urls', () => {
```
Comment (bigint) [inline tests/packages/lib/truncateUrl.spec.ts:22]:
```suggestion
  test('should not truncate *.lenster.xyz urls', () => {
```

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

Rad LGTM

**Pull request merged**